File: js/src/maplestory/api/dto/character/characterStatDto.ts
import { CharacterStatDtoBody } from '../../response/character/characterStatDtoBody';

/**
 * 캐릭터 스탯 정보
 */
class CharacterStatDto {
  /**
   * 조회 기준일
   */
  date: Date;

  /**
   * 캐릭터 직업
   */
  characterClass: string;

  /**
   * 현재 스탯 정보
   */
  finalStat: {
    statName: string;
    statValue: string;
  }[];

  /**
   * 잔여 AP
   */
  remainAP: number;

  constructor(obj: CharacterStatDtoBody) {
    const { date, character_class, final_stat, remain_ap } = obj;

    this.date = new Date(date);
    this.characterClass = character_class;
    this.finalStat = final_stat.map((stat) => {
      return {
        statName: stat.stat_name,
        statValue: stat.stat_value,
      };
    });
    this.remainAP = remain_ap;
  }
}

// 외부에서 재사용 가능하도록 export
export { CharacterStatDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterStatDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.google.gson.annotations.SerializedName;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

import java.time.ZonedDateTime;
import java.util.List;

/**
 * Character comprehensive stats information
 */
@AllArgsConstructor
@Data
@ToString
public class CharacterStatDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterStatDTO<CharacterFinalStatDTO> {

    /**
     * Reference date for query (SGT, daily data with hours and minutes set to 0)
     */
    @SerializedName("date")
    private String date;

    /**
     * Character job
     */
    @SerializedName("character_class")
    private String characterClass;

    /**
     * Current stat information
     */
    @SerializedName("final_stat")
    private List<CharacterFinalStatDTO> finalStat;

    /**
     * Remaining AP
     */
    @SerializedName("remain_ap")
    private Integer remainAp;

    /**
     * Reference date for query (SGT, daily data with hours and minutes set to 0)
     */
    public ZonedDateTime getDate() {
        return date != null
                ? Utils.toZonedDateTime(this.date)
                : null;
    }
}
